pkg/pillar/cmd/zedagent/handlelocation.go

```diff
@@ -148,15 +152,13 @@ func publishLocationToController(locInfo *info.ZInfoLocation, iteration int) {
 
 	const bailOnHTTPErr = false
 	const withNetTrace = false
-	ctxWork, cancel := zedcloud.GetContextForAllIntfFunctions(zedcloudCtx)
-	defer cancel()
-	rv, err := zedcloud.SendOnAllIntf(ctxWork, zedcloudCtx, infoURL,
-		size, buf, iteration, bailOnHTTPErr, withNetTrace)
-	if err != nil {
-		// Hopefully next timeout will be more successful
-		log.Errorf("publishLocationToController: failed (status %d): %v", rv.Status, err)
-		return
-	}
+	key := "location:" + devUUID.String()
+
+	// Even for the controller destination we can't stall the queue on error,
+	// because this is recurring call, so set @forcePeriodic to true
+	forcePeriodic := true
+	queueInfoToDest(ctx, dest, key, buf, size, bailOnHTTPErr, withNetTrace,
+		forcePeriodic, info.ZInfoTypes_ZiLocation)
 }
 
 func publishLocationToLocalServer(ctx *getconfigContext, locInfo *info.ZInfoLocation) {
```

Comment:
Since you are now also refactoring a bit, would you rename this to `publishLocationToLPS`? Also any other occurrence of LocalServer can be renamed to LPS to avoid any confusion.

Reply:
@milan-zededa Sure? 'LocalServer' is used in many places all over the code. I can, but won't this only increase confusion if I change only the function name?

Reply:
I wouldn't mind if you renamed all of it :)

Reply:
Not in this PR. Here I offer only one function rename! :)